spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/Slf4jCurrentTraceContext.java

```diff
@@ -66,9 +66,21 @@ public static Slf4jCurrentTraceContext create(CurrentTraceContext delegate) {
 	}
 
 	@Override public Scope newScope(@Nullable TraceContext currentSpan) {
+		TraceContext previousSpan = get();
```

Comment:
the call site into this should already be using maybeScope, which almost identically replicates the logic here..

Reply:
actually `maybeScope` is not being used, In recent commits, I have modified the code to use `maybeScope`.